ui/src/components/Issuer/IssueAsset.tsx

```diff
@@ -21,15 +27,27 @@ const IssueAsset = () => {
     const [ quantityPrecision, setQuantityPrecision ] = useState<string>('')
     const [ description, setDescription ] = useState<string>('')
     const [ isPublic, setIsPublic ] = useState<boolean>(true)
-    const [ observersList, setObserversList ] = useState<string>('');
+    const [ observers, setObservers ] = useState<string[]>([]);
     const [ error, setError ] = useState<ErrorMessage>();
     const [ loading, setLoading ] = useState(false);
 
+    const allRegisteredParties = [
+            useStreamQueryAsPublic(RegisteredCustodian).contracts
+                .map(ri => ({ contractId: ri.contractId, contractData: ri.payload.custodian })),
+            useStreamQueryAsPublic(RegisteredIssuer).contracts
+                .map(ri => ({ contractId: ri.contractId, contractData: ri.payload.issuer })),
+            useStreamQueryAsPublic(RegisteredInvestor).contracts
+                .map(ri => ({ contractId: ri.contractId, contractData: ri.payload.investor })),
+            useStreamQueryAsPublic(RegisteredExchange).contracts
+                .map(ri => ({ contractId: ri.contractId, contractData: ri.payload.exchange })),
+            useStreamQueryAsPublic(RegisteredBroker).contracts
+                .map(ri => ({ contractId: ri.contractId, contractData: ri.payload.broker }))
+            ].flat()
```

Comment:
idk if this is technically "legal"

Reply:
but if it is, then an issuer can add observers to their asset via dropdown

Reply:
Looks fine to me, you're just gathering up all the publicly listed parties on the ledger :)